opencog/atoms/base/Atom.h

```diff
@@ -173,6 +173,11 @@ class Atom
 
     virtual ~Atom();
 
+    virtual const std::string& getName() const {
+        static std::string empty_string("");
+        return empty_string;
```

Comment:
We could always just return a default name, i.e. "unnamed Atom" or something like that.

I generally try to avoid having to do casts and instance checks as they're slower than the virtual function overhead and they clutter the code with dependencies which make it harder to make architectural changes to the class hierarchy, though in this case, that's not likely given the difference between Nodes and Links.

The issue here is that the Python Atom doesn't know about the class hierarchy of Nodes and Links. So I could special-case Python to do an instance check or catch the exception, however, that will slow down the normal case quite a bit and we'd still have to either: force the Python caller to deal with exceptions or instance check.

For something like getting the name, that seems like overkill. Names are generally used for either printing or lookup, so an empty name won't go unnoticed in new code. Existing code already does an instance check, I didn't take any out. And since the call was not virtual before, no one could have been using it on non-Node Atoms.